cli/src/tasks/copy.ts

```diff
@@ -70,13 +72,38 @@ export async function copy(
       'capacitor:copy:before',
     );
 
+    const allPlugins = await getPlugins(config, platformName);
+    let isFederated = false;
+    if (
+      allPlugins.filter(plugin => plugin.id === '@capacitor/federation')
+        .length > 0
+    ) {
+      isFederated = true;
+    }
+
     if (platformName === config.ios.name) {
-      await copyWebDir(config, await config.ios.webDirAbs);
+      if (isFederated) {
+        await copyFederatedWebDirs(config, await config.ios.webDirAbs);
+      } else {
+        await copyWebDir(
+          config,
+          await config.ios.webDirAbs,
+          config.app.webDirAbs,
+        );
+      }
       await copyCapacitorConfig(config, config.ios.nativeTargetDirAbs);
       const cordovaPlugins = await getCordovaPlugins(config, platformName);
       await handleCordovaPluginsJS(cordovaPlugins, config, platformName);
     } else if (platformName === config.android.name) {
-      await copyWebDir(config, config.android.webDirAbs);
+      if (isFederated) {
+        await copyFederatedWebDirs(config, await config.android.webDirAbs);
```

Comment:
I noticed here that we're awaiting here and not on line 103, did `config.android.webDirAbs` become async?

Reply:
Nice catch, thanks